### DISCUSSION TITLE
how can i start Integrated Node GLSP Server

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-glsp/glsp#1143] [Q&A] [answered]
i have already start following two glsp server with vscode.
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/9fa93e68-e805-4750-9dbc-cbe291f0b134)
but how can i start Integrated Node GLSP Server,when i debug code ,it seems that something go wrong in following code
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/2f084b63-31c9-4df6-a7af-f74af06f1352)
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/559d5d34-9666-429e-bc0c-d022ff0dbdb2)
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/6bfd41b4-c18e-49e6-92bf-f8779c2f065e)

**Comment by rsoika** [comment]
hi @zrwsmd, just one general comment from my side as you post a lot in the last days which is fine. 
Please describe your problems in more detail using words and not using screenshots. Did you know that you can post also code examples in github by putting it into a block starting and ending with this ```   - see also [official guideline here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code).

If you take yourself more time to describe a problem, it makes it easier to others to help you.

From you screenshot it looks like you are trying [vscode-integration of glsp](https://github.com/eclipse-glsp/glsp-vscode-integration)? Did you already have a glsp server and client implemented? You need to start the glsp server part before you can test a vscode-integration as an extension.

**Reply by zrwsmd** [reply]
i want to start a node glsp server,i guess it is related to NodeGlspVscodeServer  which is in glsp-vscode-integration-master\example\workflow\extension\src\workflow-extension.ts,but when i choose Workflow GLSP Example Extension (Integrated Node GLSP Server) launch option and start it with debug mode,  i found it here who has error(below scrennshot) ,below screenshot "workflowServer"  is a instance of NodeGlspVscodeServer
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/febcb25e-bb34-4a27-bceb-31212bd68c94)
debug inside of it ,it here that has error
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/2f084b63-31c9-4df6-a7af-f74af06f1352)

**Reply by rsoika** [reply]
Hi @zrwsmd  it is still not clear what you are trying to do. 
In general GLSP works always with two parts: 

 - the server part is responsible to provide the model information and operations
 - the client part is responsible to show the visual modeller plane 

both parts communicate over a internal protocol. So each time you develop and test your glsp code, you need to start both parts - the server and the client.  you can test this easily with the [GLSP workflow example](https://github.com/eclipse-glsp/glsp-examples). 
The glsp workflow example uses the Eclipse Theia platform as the core IDE to show your modeller. This is a kind of default setup.
But you can integrate GLSP also in other IDEs like VS-Code or Eclipse. In this cases you need a IDE-Integration code - like the VSCode Integration project. Such a project - more or less - simply wraps you client/server parts into a IDE extension.

So the question is: what do you try to archive? Do you what to develop a modeller or do you already have developed you modeller and integrate it into an additional IDE?

**Reply by zrwsmd** [reply]
i just want to start  internal node glsp server inside (https://github.com/eclipse-glsp/glsp-vscode-integration) project,but currently something error like
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/008868d7-3a0b-4389-a0da-57beb1d0a73f)

but i can launch external glsp server using external java jar ,and then start webview like below screenshot
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/40697f16-43ec-47b5-951b-1fc371a65ddc),

how can i achieve to start like above screenshoot with internal node glsp server
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/840045f3-c493-4383-b518-0f33f3f34fab)

**Reply by rsoika** [reply]
ok, so to sum it up: 

You can't start the [glsp-vscode-integration](https://github.com/eclipse-glsp/glsp-vscode-integration) with the integrated nodejs server using the vscode lauch option. But launching the server external and the vscode extension separately works - correct?

If you check the log files this will help to narrow down the problem further.

**Reply by zrwsmd** [reply]
yes,so where can i check  log file ,i found error is line 49 ,i guess is GLSPServer instance can not fetch from DI container  
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/3551fced-e2a0-44f6-b413-30804126f289),but container(this.serverContainer)
seems  have this instance
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/b05c1bfd-e33f-417b-9211-25cf558b8053)
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/ad0dcdb2-4268-492a-98c7-96b55b642d39)

**Reply by rsoika** [reply]
Once again: If you copy the error message as text into this discussion (marked as a code fragment) we all could read the full error message. If you copy a screenshot this is not very helpful.

**Reply by zrwsmd** [reply]
the above line 49 have following errors
Error: No matching bindings found for serviceIdentifier: Symbol(GLSPServer)
	at _validateActiveBindingCount (D:\glsp-vscode-integration-master\node_modules\inversify\lib\planning\planner.js:63:23)
	at _getActiveBindings (D:\glsp-vscode-integration-master\node_modules\inversify\lib\planning\planner.js:49:5)
	at _createSubRequests (D:\glsp-vscode-integration-master\node_modules\inversify\lib\planning\planner.js:86:26)
	at Object.plan (D:\glsp-vscode-integration-master\node_modules\inversify\lib\planning\planner.js:137:9)
	at D:\glsp-vscode-integration-master\node_modules\inversify\lib\container\container.js:333:37
	at Container._get (D:\glsp-vscode-integration-master\node_modules\inversify\lib\container\container.js:326:44)
	at Container.get (D:\glsp-vscode-integration-master\node_modules\inversify\lib\container\container.js:243:21)
	at NodeGlspVscodeServer.createGLSPClient (D:\glsp-vscode-integration-master\packages\vscode-integration\lib\node\quickstart-components\node-glsp-vscode-server.js:44:45)
	at NodeGlspVscodeServer.start (D:\glsp-vscode-integration-master\packages\vscode-integration\lib\common\quickstart-components\base-glsp-vscode-server.js:45:43)
	at activate (d:\glsp-vscode-integration-master\example\workflow\extension\lib\workflow-extension.js:101:20)
	at q.ib (d:\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:136:13192)
	at q.hb (d:\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:136:12914)
	at d:\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:136:11039
	at async v.n (d:\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:125:6205)
	at async v.n (d:\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:125:6205)
	at async v.m (d:\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:125:6168)
	at async v.l (d:\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:125:5625) {stack: 'Error: No matching bindings found for service…ch\api\node\extensionHostProcess.js:125:5625)', message: 'No matching bindings found for serviceIdentifier: Symbol(GLSPServer)'}

**Reply by zrwsmd** [reply, marked as answer]
new version seems to fix it ,thanks

**Reply by rsoika** [reply]
@zrwsmd This is great!. Can you please post the version that caused the problem and the new version which solved the problem in your case? This could help other users with a similar problem.

**Reply by zrwsmd** [reply]
new version 
   "@eclipse-glsp/layout-elk": "~2.0.0",
   "@eclipse-glsp/server": "~2.0.0",
   "@eclipse-glsp/vscode-integration": "~2.0.0",
   "@eclipse-glsp/client": "~2.0.0",
in total,it 2.0.0 version which fix it


old version
    "@eclipse-glsp/layout-elk": "1.1.0-next.e49c816.56+e49c816",
    "@eclipse-glsp/server": "1.1.0-next.e49c816.56+e49c816",
     "@eclipse-glsp/vscode-integration-webview": "1.1.0-next",
	  "@eclipse-glsp/vscode-integration": "1.1.0-next",
	  "@eclipse-glsp/client": "1.1.0-next.69a77b0.259+69a77b0",
	  
in old version,node glsp server can not start with error No matching bindings found for serviceIdentifier: Symbol(GLSPServer)
